clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -1625,6 +1619,33 @@ protected TopicSwitch resolveSourceKafkaServersWithinTopicSwitch(TopicSwitch ori
     return originalTopicSwitch;
   }
 
+  protected static long getOffsetToOnlineLagThresholdPerPartition(
+      Optional<HybridStoreConfig> hybridStoreConfig,
+      ReadOnlyStoreRepository storeRepository,
+      String storeName,
+      int versionNumber,
+      int subPartitionCount) {
+    if (!hybridStoreConfig.isPresent()) {
+      throw new VeniceException("This is not a hybrid store: " + storeName);
+    }
+    long lagThreshold = hybridStoreConfig.get().getOffsetLagThresholdToGoOnline();
+    if (lagThreshold < 0) {
+      /**
+       * Offset lag threshold is disabled, use time lag.
+       */
+      return lagThreshold;
+    }
+    boolean isBackupVersion = versionNumber < storeRepository.getStore(storeName).getCurrentVersion();
+
+    /**
+     * For the backup version, we would like to speed up the ready-to-serve check during restart, and
+     * it will be using configured `OffsetLagThresholdToGoOnline` in partition-level.
+     */
+    return isBackupVersion
+        ? hybridStoreConfig.get().getOffsetLagThresholdToGoOnline()
+        : lagThreshold / subPartitionCount;
```

Comment:
I remember we used to have off by 1 issues around lag calculation so we advised users to not set lag threshold to 0. Are we all clear with that? Since this division might return `0` for user stores with large amount of partitions. e.g. 2000 partitions and lag threshold of 1000. Might be worth to have an integration test to validate the 0 lag threshold scenario.

Alternatively we can also set our own lower limit of like 1 for partition level offset lag threshold.

Reply:
Good question, and I am not aware of this `0` lag issue.
Let us sync up internally.